ctapipe/visualization/mpl_array.py

```diff
@@ -202,7 +202,7 @@ def set_vector_rho_phi(self, rho, phi, c=None, **kwargs):
         uu, vv = polar_to_cart(rho, phi)
         self.set_vector_uv(uu, vv, c=c, **kwargs)
 
-    def set_vector_hillas(self, hillas_dict, length, time_gradient, angle_offset):
+    def set_vector_hillas(self, event, hillas_dict, length, time_gradient, angle_offset):
```

Comment:
This seems to be overkill to pass the whole event, and also then you repeat the hillas and timing parameters (which are in the event as well) .  I would suggest just adding another parameter `core_dict`, so it's similar to `hillas_dict` and `time_gradient`, and then isn't dependent on the event structure.   Also the time_gradient is also a dict, but the name of the variable makes it look like a scalar compared to the others.

So the call signature should be like this:
```python3
def set_vector_hillas(
    self, 
    hillas_dict : Dict[int, HillasParameters], 
    core_dict: Dict[int, CoreParameters],  
    time_gradient_dict : Dict[int, float], 
    length,
    angle_offset
):
```

Otherwise, you can just pass the event only and remove hillas_dict,  `time_gradient`
```python3
def set_vector_hillas(self, event):
```

but I don't really like to have these low-level plotting functions relying on the event structure, since it's then very hard to see what data they are actually using (the event contains everything from R0 on up).

Reply:
What you refer to is not a dict, but rather a single value per tel_id coming from`event.dl1.tel[tel_id].parameters.core.psi`.
Inside each of the two methods, there is then a cycle over tel_ids.

Do you mean I should do something like this before the call to the ArrayDisplay method(s)?

```python
core_dict = {
            tel_id: dl1.parameters.core.psi
            for tel_id, dl1 in event.dl1.tel.items()
        }
```

Reply:
Yes, then it would be equal for all parameters.

Reply:
It is suprising that some parameters are taken from event and some are required to be passed in as a mapping `tel_id → parameter`. It should be equal for all parameters.

Reply:
what do you mean?

Reply:
Right now, you pass the full event and some dicts. All information that is in the dicts is also in `event`. That does not make sense.

Either you only pass `event` and get all info from there, or you pass each info you need individually. But a mixture is not good.

Reply:
Ah yes sure, I will just pass the dictionary as I showed here